doc/sphinx/proof-engine/tactics.rst

```diff
@@ -898,211 +1109,156 @@ Applying theorems
+      non-dependent premises.
 
-      This repeats :tacn:`intro` until the :token:`natural`\-th non-dependent
-      product.
+      .. todo PR: should we discourage use of "until" because it lets Coq pick the names?
+         (maintainability issue)
+         Also, shall I make "intros until" a separate command?
 
       .. example::
 
-         On the subgoal :g:`forall x y : nat, x = y -> y = x` the
-         tactic :n:`intros until 1` is equivalent to :n:`intros x y H`,
-         as :g:`x = y -> y = x` is the first non-dependent product.
+         On the goal :g:`forall x y : nat, x = y -> y = x` the
+         tactic :n:`intros until 1` is equivalent to :n:`intros x y H`
+         because :g:`x = y -> y = x` is the first non-dependent premise.
+
+         .. todo PR: think I should drop the following example.  z is not
+            used on the RHS--unless we can change it so its useful
 
-         On the subgoal :g:`forall x y z : nat, x = y -> y = x` the
+         On the goal :g:`forall x y z : nat, x = y -> y = x` the
          tactic :n:`intros until 1` is equivalent to :n:`intros x y z`
          as the product on :g:`z` can be rewritten as a non-dependent
-         product: :g:`forall x y : nat, nat -> x = y -> y = x`.
+         premise: :g:`forall x y : nat, nat -> x = y -> y = x`.
 
       .. exn:: No such hypothesis in current goal.
 
          This happens when :token:`natural` is 0 or is greater than the number of
          non-dependent products of the goal.
 
-   .. tacv:: intro {? @ident__1 } after @ident__2
-             intro {? @ident__1 } before @ident__2
-             intro {? @ident__1 } at top
-             intro {? @ident__1 } at bottom
-
-      These tactics apply :n:`intro {? @ident__1}` and move the freshly
-      introduced hypothesis respectively after the hypothesis :n:`@ident__2`,
-      before the hypothesis :n:`@ident__2`, at the top of the local context,
-      or at the bottom of the local context. All hypotheses on which the new
-      hypothesis depends are moved too so as to respect the order of
-      dependencies between hypotheses. It is equivalent to :n:`intro {? @ident__1 }`
-      followed by the appropriate call to :tacn:`move`,
-      :tacn:`move … before …`, :tacn:`move … at top`,
-      or :tacn:`move … at bottom`.
-
-      .. note::
-
-         :n:`intro at bottom` is a synonym for :n:`intro` with no argument.
-
-      .. exn:: No such hypothesis: @ident.
-         :undocumented:
-
-.. tacn:: intros {* @intropattern }
-   :name: intros …
-
-   Introduces one or more variables or hypotheses from the goal by matching the
-   intro patterns.  See the description in :ref:`intropatterns`.
-
 .. tacn:: eintros {* @intropattern }
-   :name: eintros
 
-   Works just like :tacn:`intros …` except that it creates existential variables
+   Works just like :tacn:`intros` except that it creates existential variables
    for any unresolved variables rather than failing.
```

Comment:
Typically, this can happen when using the `%` intro pattern (I don't know of any other use case).